pkg/odo/util/validation/validatable.go

```diff
@@ -0,0 +1,13 @@
+package validation
+
+// Validatable represents a common ancestor for validatable parameters
+type Validatable struct {
+	Required             bool        `json:"required,omitempty"`
```

Comment:
Can you please add some comments around what each of these mean?
But awesome code I must say... remembered by Java days :)